### DISCUSSION TITLE
How might I separate renovate configs on the same repo?

### DISCUSSION THREAD

**Discussion by patsplat** [renovatebot/renovate#25477] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Latest npm

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I am trying to configure renovatebot to auto batch downstream services when various upstream packages are built.  I hope to run renovate against the downstream service in the CI/CD for the various cross cutting upstream packages.

On any upstream release build, I’d like renovate to make sure there is one open MR patching to latest in the downstream service.  Different upstream libs should have different MRs, but no more than one MR per upstream dependency.

Thus I would like to be able to configure renovatebot to run differently for separate packages on the same repo on separate runs of the cli.

### What you tried so far.

For the first package, I was able to get the package rules to target the first upstream package and create the first merge request.  

However when I ran it a second time targeting the second upstream package, it closed the MR from the first run.

It feels slow close to working perfectly (thanks!) but it seems like I am missing something.

How can I keep the renovate running for package A to avoid messing with the MRs created by the renovate running for package B?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by patsplat** [comment]
Is there some kind of config allowing multiple renovates to coexist in the same repo?

or is the model one renovatebot running per repo, with various package configs?

how much flexibility is there?

**Comment by rarkins** [comment]
If you want to change config between runs then you need to do it with global config and not repo config. You also should look into pruneStaleBranches=false

**Comment by viceice** [comment]
instead of using different configs, use appropriate package rules.

**Comment by patsplat** [comment]
Thank you for the responses.

As I said, I'm trying to setup ci/cd for an internal push model. The upstream internal library build immediately pushes the patches out to the downstream internal service repos.

From what I'm hearing, it sounds like the way to achieve this goal is to configure renovate with appropriate packagerules to pull the internal patches. The push behavior can then be achieved by the upstream build kicking off the pulling renovate job.
